# HG changeset patch
# User prr
# Date 1342804653 25200
#      Fri Jul 20 10:17:33 2012 -0700
# Node ID 84d5cdbe7bce2caf907ec842ac7b4fa2097fb295
# Parent  bf5bf37108c14d1d01b8a1043f9e3ff0e3727869
7183251: Netbeans editor renders text wrong on JDK 7u6 build
Reviewed-by: igor, jgodinez

diff -r bf5bf37108c1 -r 84d5cdbe7bce src/share/classes/sun/font/SunLayoutEngine.java
--- openjdk/jdk/src/share/classes/sun/font/SunLayoutEngine.java	Fri Jul 20 13:58:52 2012 +0400
+++ openjdk/jdk/src/share/classes/sun/font/SunLayoutEngine.java	Fri Jul 20 10:17:33 2012 -0700
@@ -137,8 +137,9 @@
 
         LayoutEngine e = (LayoutEngine)cache.get(key);
         if (e == null) {
-            e = new SunLayoutEngine(key.copy());
-            cache.put(key, e);
+            LayoutEngineKey copy = key.copy();
+            e = new SunLayoutEngine(copy);
+            cache.put(copy, e);
         }
         return e;
     }
